File: javaProblems/arrays_lists/ClosestMinMax.java
package arrays_lists;

public class ClosestMinMax {
    public int solve(int[] A) {
        int N = A.length;
        int min = A[0], max = A[0];
        for (int j : A) {
            if (j > max) max = j;
            if (j < min) min = j;
        }
        if (max == min){
            return 1;
        }
        int last_min_index = -1;
        int last_max_index = -1;
        int ans = N;
        for(int i=0;i<N;i++){
            if (A[i] == min) last_min_index = Math.max(i, last_min_index);
            if (A[i] == max) last_max_index = Math.max(i, last_max_index);
            if (last_min_index != -1 && last_max_index != -1){
                ans = Math.min(ans, Math.abs(last_max_index-last_min_index)+1);
            }
        }
        return ans;
    }
}

File: pyProblems/arrays/carry_forward/closest_min_max.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def solve(self, A):
        last_min_index = -1
        last_max_index = -1
        ans = len(A)
        min_element = min(A)
        max_element = max(A)
        for i in range(len(A)):
            if A[i] == min_element:
                last_min_index = i
                if last_max_index != -1:
                    ans = min(ans, last_min_index - last_max_index + 1)

            if A[i] == max_element:
                last_max_index = i
                if last_min_index != -1:
                    ans = min(ans, last_max_index - last_min_index + 1)
        return ans

